mlir/lib/Conversion/TensorToSPIRV/TensorToSPIRV.cpp

```diff
@@ -45,6 +45,9 @@ class TensorExtractPattern final
                   ConversionPatternRewriter &rewriter) const override {
     auto tensorType = cast<RankedTensorType>(extractOp.getTensor().getType());
 
+    if (!tensorType.getElementType().isIntOrFloat())
+      return rewriter.notifyMatchFailure(extractOp,
+                                         "only integers and floats supported");
```

Comment:
This looks fine, but I wonder if this should really be `if (!isa<spirv::ScalarType>(tensorType.getElementType))` to also reject invalid integer/float types? For example `i128` shouldn't be supported.

Reply:
Good idea, I'll fix it.